apps/node/src/app/main/routes/auth.py

```diff
@@ -0,0 +1,43 @@
+from functools import wraps
+from json import dumps
+
+from flask import Response, request
+from flask import current_app as app
+from syft.codes import RESPONSE_MSG
+import jwt
+
+from ..core.exceptions import MissingRequestKeyError, InvalidCredentialsError
+from ..users import User
+from ... import db
+
+
+def token_required(f):
```

Comment:
Why can't we share this logic between here and `app/main/events/auth.py`?

Reply:
The location  of the token changes from http to websocket so the function's logic is not quite the same

Reply:
Also, the way the exceptions are handled also change, in one of them you gotta create a Response object, while in the other you simply return a object as a string

Reply:
I've done it via [decorator factory](https://github.com/OpenMined/PyGrid/pull/708/files#diff-fe0006987fa850f55246e88f8dcb04ec)

![image](https://user-images.githubusercontent.com/9937551/92424537-023b9a80-f15b-11ea-9190-daca84362a46.png)